part-2.py

```diff
@@ -4,13 +4,38 @@
 # the appropriate comment.
 
 # search
-
-
+def search(array,query):
+    if array == []:
+        return False
+    if array[0] == query:
+        return True
+    x = array[0]
+    array.remove(array[0])
+    return search(array, query)
 
 # is_palindrome
-
-
+def is_palindrome(text):
+    if len(text) <= 1:
+        return True
+    if text[:1] == text[-1:]:
+        return is_palindrome(text[1:-1])
+    else:
+        return False
 
 # digit_match
-
+# note that this copies Kaida's code-along, as I had some issues
+# I actually did get pretty close though!
+
+def digit_match(x,y, matches=0):
```

Comment:
Great work working through this tricky problem. A way to refactor and DRY up your code is to convert `x % 10 == y % 10` to a 0 or 1 using  `int(x % 10 == y % 10)`, that way you can return `matches + int(x % 10 == y % 10)` and don't require and if/else.